cluster/gce/config-default.sh

```diff
@@ -562,10 +562,21 @@ export CLOUD_PROVIDER_FLAG="${CLOUD_PROVIDER_FLAG:-external}"
 export ENABLE_AUTH_PROVIDER_GCP="${ENABLE_AUTH_PROVIDER_GCP:-false}"
 
 # External cloud provider requires ENABLE_AUTH_PROVIDER_GCP and feature flags
-# DisableKubeletCloudCredentialProviders and DisableCloudProviders
+# DisableKubeletCloudCredentialProviders and DisableCloudProviders should be set to true
 if [[ "${CLOUD_PROVIDER_FLAG:-}" == "external" ]]; then
+  echo "before FEATURE_GATES: ${FEATURE_GATES:-}"
+  if [[ "${FEATURE_GATES:-}" == *"DisableKubeletCloudCredentialProviders=false"*  ]]; then
+    echo -e "${color_red:-}ERROR: Cannot set DisableKubeletCloudCredentialProviders to false when CLOUD_PROVIDER_FLAG is set to external." >&2
+  fi
+  if [[ "${FEATURE_GATES:-}" == *"DisableCloudProviders=false"*  ]]; then
+    echo -e "${color_red:-}ERROR: Cannot set DisableCloudProviders to false when CLOUD_PROVIDER_FLAG is set to external." >&2
+  fi
   export ENABLE_AUTH_PROVIDER_GCP=true
-  if [[ -n "${FEATURE_GATES:-DisableKubeletCloudCredentialProviders=True,DisableCloudProviders=True}" ]]; then
-    export FEATURE_GATES="${FEATURE_GATES},DisableKubeletCloudCredentialProviders=True,DisableCloudProviders=True"
+  if [[ "${FEATURE_GATES:-}" != *"DisableKubeletCloudCredentialProviders=true"*  ]]; then
+    export FEATURE_GATES="${FEATURE_GATES},DisableKubeletCloudCredentialProviders=true"
```

Comment:
i'm noticing some output like this:
```
FEATURE_GATES: ,DisableKubeletCloudCredentialProviders=true,DisableCloudProviders=true
```
do we need to have some protection to ensure that the leading comma (`,`) is not created when `FEATURE_GATES` is empty?

Reply:
I think that is the problem, IIRC there is some bash magic in one of those files that handles that comma separated strings format